File: Leetcode_solutions/Pascal'sTriangle.js
/**
 * @param {number} numRows
 * @return {number[][]}
 */
var generate = function(numRows, row = 0, column = 1, output = [[1]]) {
  // i row
  // j column
  // base
  if (numRows === 0) {
    return [];
  }
  if (row > numRows) {
    return output;
  } else {
    if (row >= output.length) {
      output.push([]);
      output[row].push(1);
      for (let i = 1; i < row; i++) {
        output[row].push(output[row - 1][i - 1] + output[row - 1][i]);
      }
      output[row].push(1);
    }
    return generate(numRows, row + 1, column++, output);
  }
};
console.log(generate(3));
console.log(generate(5));

File: Leetcode_solutions/Pascal's Triangle.py
class Solution:
    def generate(self, numRows: int) -> List[List[int]]:
        if numRows == 0:
            return []
        ans = [[1]]
        for i in range(1, numRows):
            current = []
            for j in range(i + 1):
                if j == 0 or j == i:
                    current.append(1)
                else:
                    current.append(ans[-1][j] + ans[-1][j - 1])
            ans.append(current)
        return ans
